src/Illuminate/Support/Sleep.php

```diff
@@ -448,6 +448,38 @@ public function unless($condition)
         return $this->when(! value($condition, $this));
     }
 
+    /**
+     * Sleep while the closure evaluates to falsey.
+     *
+     * @template TReturn of mixed
+     *
+     * @param  (callable(int): TReturn)  $callback
+     * @param  int  $bailAfter
+     * @return TReturn
+     */
+    public function while($callback)
+    {
+        $this->shouldNotSleep();
+
+        $sleeps = 0;
+
+        while (true) {
```

Comment:
What happens if the $result is not truthy? Does it run forever and ignore the sleep duration? 

`while (true)` are you sure?